### PR TITLE
Submitting the BZZ Stream Protocol and its diagrams

### PR DESCRIPTION
**Pull request by FantasticoFox**

### CODE REVIEW

**Comment by nagydani** [comment]
The text (md) of the SWIP is missing, only PNG assets have been included. Please include the text as well.

**Comment by FantasticoFox** [comment]
Can you check again?

**Comment by nagydani** [comment]
> Can you check again?

Now it is there. Thank you!
The proposed SWIP is good enough for number 10 to be assigned to it: both the addressed concern and the way it is addressed merit a SWIP and the formatting is sufficiently good. So, please rename all the files in accordance with SWIP-0 and change the file header so that both reflect the fact that it will be SWIP-10. Then I will continue to review the substance. Use `git mv` for renaming.

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+<!--You can leave these HTML comments in your merged SWIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SWIPs. Note that a SWIP number will be assigned by an editor. When opening a pull request to submit your SWIP, please use an abbreviated title in the filename, `SWIP-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+This is the suggested template for new SWIPs.
+
+Note that a SWIP number will be assigned by an editor. When opening a pull request to submit your SWIP, please use an abbreviated title in the filename, `SWIP-draft_title_abbrev.md`.
+The title should be 44 characters or less.
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SWIP.-->
+If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SWIP.
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Text from SWIP template, please remove it.

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+author: Viktor Trón, (@zelig) <viktor@ethswarm.org>,  Tim Bansemer (editorial) (@FantasticoFox) <tim@ethswarm.org>, Elad Nachmias, (@acud) <elad@ethswarm.org>
+
+status: Draft
+type: Standards Track Core, Networking
+category: Core | Networking
+created: 2019-08-01
+---
+
+<!--You can leave these HTML comments in your merged SWIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SWIPs. Note that a SWIP number will be assigned by an editor. When opening a pull request to submit your SWIP, please use an abbreviated title in the filename, `SWIP-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+This is the suggested template for new SWIPs.
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Text from SWIP template, please remove it.

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SWIP.-->
+If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SWIP.
+
+The BZZ Stream Protocol facilitates data transmission between two swarm nodes, specifically targeting sequential data in the form of a sequence of chunks as defined by swarm.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+A short (~200 word) description of the technical issue being addressed.
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Text from SWIP template, please remove it.

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+The protocol should cater for the following requirements:
+
+ * client should be able to request arbitrary ranges of hashes from the serving peer
+ * client can be assumed to have some of the data already and therefore can opt in to selectively request chunks based on their hashes
+
+To mitigate duplicate data transmission the stream protocol provides a configurable message roundtrip before executing a batch delivery which allows the downstream peer to selectively request the chunks which it does not store at the time of the request. 
+
+When delivery batches are pre-negotiated (i.e. when the client selectively tells the server which chunks it would like to receive), we can conclude that the delivery batches are optimsed for urgency rather than for maximising batch utilisation (since the server sends a certain batch that potentially gets reduced into a smaller one by the client before actually being transmitted)
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Minor typo: optimsed

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+GetRange (unbounded) - client requests an unbounded range (specifies only `From` parameter)<br>
+![unbounded-range](stream-unbounded.png)
+<br>
+GetRange (no roundtrip) - client requests an unbounded or bounded range with no roundtrip configured<br>
+![unbounded-range](stream-no-roundtrip.png)
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+There are several occasions for nodes to exchange data with each other where the data is defined of a squence of chunks of Swarm nodes. There are significant bandwith savings by having a formalised protocol functionality which allows to check for offered hashes and already available hashes of existing chunks for the retrieving node.
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Minor typo: squence

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+Example 1: 
+A pull sync
+ * Simultanous streams
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Minor typo: Simultanous

File: SWIPs/Swip-10 BZZ Stream Protocol [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+While they compete, they will save bandwith by only requested specfic chunks still due for transmission instead of full hash-ranges.
+
+## Backwards Compatibility
+<!--All SWIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The SWIP must explain how the author proposes to deal with these incompatibilities. SWIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for SWIPs that are affecting changes to data and message formats. Other SWIPs can choose to include links to test cases if applicable.-->
+
+Test cases for the Go-Client:
+[TODO] for summarize high-level test szenarios which need to be covered to proof that the specc works as fledged out.
```
Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Minor typo: szenarios

Comment (diegomasini) [inline SWIPs/Swip-10 BZZ Stream Protocol]: Minor typo: specc

**Pull request closed without merging**